prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -207,11 +212,24 @@ object CustomFHIRFunctions : FhirPathFunctions {
         }
 
         var timePeriodSpecified: String? = null
-        var comparisonDate: LocalDate = LocalDate.now()
-        if (!parameters.isNullOrEmpty() && !parameters[0].isNullOrEmpty()) {
-            val params = getParamValue(parameters[0])
-            timePeriodSpecified = params.second
-            comparisonDate = params.first
+        var comparisonDate: LocalDate? = LocalDate.now()
+        if (!parameters.isNullOrEmpty()) {
+            if (parameters.size > 2) {
+                throw SchemaException("Must call the convertDateToAge function with no more than two parameters")
+            }
```

Comment:
Maybe something like this too? Might make these checks long enough to justify a helper function again, idk 😬 
```suggestion
if (parameters.size > 2) {
                throw SchemaException("Must call the convertDateToAge function with no more than two parameters")
            } else if ((parameters.size > 0 && parameters[0].size != 1) ||
                (parameters.size > 1 && parameters[1].size != 1)
            ) {
                throw SchemaException("Must call the convertDateToAge function without empty parameters")
            } else if (parameters.size == 2) {
                if ((parameters[0][0] is DateType && parameters[1][0] is DateType) ||
                    (parameters[0][0] is StringType && parameters[1][0] is StringType)
                ) {
                    throw SchemaException("Cannot call the convertDateToAge function with two parameters of the same type")
                }
            }
```

Reply:
Got it. Added checks for that as well as tests.